python/tskit/trees.py

```diff
@@ -5090,13 +5149,13 @@ def variants(
             isolated_as_missing=isolated_as_missing,
             alleles=alleles,
         )
-        sites = range(self.num_sites)
+        start, stop = np.searchsorted(self.tables.sites.position, interval)
```

Comment:
We're incurring a full copy of the tables here even when we don't specify ``left`` or ``right`` - could have been easily avoided.

Reply:
Good point. I'll code around it.